src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceIdentifier.cs

```diff
@@ -138,10 +138,18 @@ public override string ToString()
 
             AppendIfNotNull(ref resourceId, "/subscriptions/{0}", Subscription);
             AppendIfNotNull(ref resourceId, "/resourceGroups/{0}", ResourceGroupName);
-            AppendIfNotNull(ref resourceId, "/providers/{0}", provider);
-            AppendIfNotNull(ref resourceId, "/{0}", parentAndType);
-            AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
 
+            // Only append parentAndType and ResourceName when provider is not empty.
+            if (!string.IsNullOrEmpty(provider))
+            {
+                resourceId.AppendFormat("/providers/{0}", provider);
+                if (!string.IsNullOrEmpty(parentAndType))
+                {
+                    resourceId.AppendFormat("/{0}", parentAndType);
+                    AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
+                }
+            }
```

Comment:
reverted the change in this file.

---
In reply to: [118760265](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118760265) [](ancestors = 118760265,118643432)